pkgs/tools/graphics/diagrams-builder/default.nix

```diff
@@ -0,0 +1,43 @@
+/*
+  If user need access to more haskell package for building his
+  diagrams, he simply has to pass these package through the
+  extra packages function as follow in `config.nix`:
+  
+  ~~~
+  diagrams-builder.override {
+    extraPackages = self : [myHaskellPackage];
+  }
+  ­~~~
+*/
+
+{ stdenv, ghcWithPackages, makeWrapper, diagrams-builder, extraPackages ? (self: []) }:
+let
+  # Used same technique as for the yiCustom package.
+  w = ghcWithPackages 
+    (self: [ diagrams-builder ] ++ extraPackages self);
+  wrappedGhc = w.override { ignoreCollisions = true; };
+  ghcVersion = w.version;
+in
+stdenv.mkDerivation {
+  name = "diagrams-builder";
+  buildInputs = [ makeWrapper ];
+  buildCommand = ''
```

Comment:
Commit c449a93 adds a `postBuild` hook to the normal `ghcWithPackages` function that can be used to wrap programs, etc. Maybe the implementation of this function can become even simpler by using that mechanism?